proto/internal/temporal/server/api/matchingservice/v1/request_response.proto

```diff
@@ -329,6 +329,17 @@ message GetBuildIdTaskQueueMappingResponse {
     repeated string task_queues = 1;
 }
 
+message ForceLoadTaskQueuePartitionRequest {
+    string namespace_id = 1;
+
+    temporal.server.api.taskqueue.v1.TaskQueuePartition task_queue_partition = 2;
```

Comment:
We already do it in `DescribeTaskQueuePartition`
```Go
message DescribeTaskQueuePartitionRequest {
    string namespace_id = 1;
    temporal.server.api.taskqueue.v1.TaskQueuePartition task_queue_partition = 2;
    ...
}
```

I couldn't find any other cases that I was sure are what you describe, except for `ForceUnloadTaskQueueRequest`, which I did have the changes in this PR, but pulled them to simplify the PR. Would you like me to put them back?
Go```
message ForceUnloadTaskQueueRequest {
    string namespace_id = 1;
    string task_queue = 2;
    temporal.api.enums.v1.TaskQueueType task_queue_type = 3;
}
```

There are a couple using `temporal.api.taskqueue.v1.TaskQueue` but they aren't referencing partitions
```Go
message CancelOutstandingPollRequest {
    string namespace_id = 1;
    temporal.api.enums.v1.TaskQueueType task_queue_type = 2;
    temporal.api.taskqueue.v1.TaskQueue task_queue = 3;
    string poller_id = 4;
}
```

I chose the object from the API proto files because it ensures that the partition specification is typed. I chose this specifically because of an error I made in development, where I called `tqid.NormalPartitionFromRpcName(req.GetTaskQueue(), req.GetNamespaceId(), req.GetTaskQueueType())` thinking that was all I needed, but on the Caller end I wasn't setting the TaskQueue string to include the partition. The typing of the RPC provides better insurance against this mistake.

Reply:
All the requests that go to matching service that are for a particular partition, except for `DescribeTaskQueuePartitionRequest`, have the name and type separate with the partition embedded in the name. I would have asked for that on `DescribeTaskQueuePartitionRequest` too but I didn't review that specific PR.

Note that for the ones that come directly from clients the task queue name (and type if present) is in the wrapped message, not in the matchingservice.XRequest directly.

CancelOutstandingPollRequest is certainly directed at a specific partition, and needs to be to get routed to the correct matching node. As with the others, the partition is embedded in the name. The implementation in matching engine used to look up the partition since the poller map was in there, but I moved it to host level.

I'm not saying the new TaskQueuePartition message isn't better, it clearly is. But it's much more important that the API be internally consistent so it's easier to form a mental model of it. There are already enough exceptions and special cases in the protos and codebase in general, let's not add more. It may be worth migrating all calls to TaskQueuePartition, but let's do it deliberately and all at once, instead of having a mix.

Reply:
**This conversation should pause until https://github.com/temporalio/temporal/pull/6324/files#r1690837324 is concluded.**


So, this may just be me coming from software shops that don't justify that kind of api improvement/unification work. Because of this, I take new development as the opportunity to move the needle in the direction that we would want the system overall. Additionally, I don't see this as a new special case, since we already have one instance of it, and are using the message on the frontend API. Finally, We both agree this is the direction the internal API should move. 

If we were migrating more deliberately, I imagine we would do it on request at a time anyways. I already have the changes stashed for another. I think this is the way we should write this new bit, and I'm willing to throw a few more PRs up behind this one if thats what you need to be willing to adopt this RPC with the improved fields.

Reply:
I don't think we would or should do it once at a time, because that would prolong things unnecessarily. We need to keep both old and new routing fields around with fallback logic for the duration of an oss release cycle (three months, these days), and I certainly wouldn't want it to drag on for > 1 cycle, so we might as well just do them all at once.

If you're committing to move the rest of them over, then this is okay.